### PR TITLE
Add Terms Modal

### LINKED ISSUE #267: Don't clear the new user's registration form when viewing terms page
Perhaps the simplest solution for this is to open the terms in a new tab with `target="_blank"` and `rel="noopener noreferrer"` attributes on the link. On the other hand, the preferable UX solution is probably to open the terms in a modal or something. If we want to do that we should consider pulling a react one if from a library e.g. http://www.material-ui.com/#/components/dialog

### PR DESCRIPTION
**Pull request by fredmon3**
This PR closes #267 

#### What does this PR do?

This PR adds a modal to display the terms so that a user's previously entered registration data is not cleared.

### CODE REVIEW

File: client/src/components/Register.js
Comment (wdoug) [inline client/src/components/Register.js]: It might be nice to pull this out into it's own component at some point. ...Note to future self

Comment (wdoug) [reply client/src/components/Register.js]: Oh we already have it in a separate component. Can we put it back?

**Comment by wdoug** [comment]
It looks like now that we have `yarn` the `yarn.lock` file needs to be upgraded for your added package.

**Comment by wdoug** [comment]
@fredmon3 I just rebased this without yarn

File: client/src/components/Register.js
Comment (wdoug) [inline client/src/components/Register.js]: Can we make this link tab accessible? If `Link`s don't support [tabindex](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/tabindex) we could use a regular `a` tag. Now that I think about it actually, I think my personal preference would be to actually have the `Link` update the url (probably as a query parameter) and then the modal is displayed based on the prop from the router query parameter. The close button could then remove that url query parameter. That way we would have an actual url for the terms still.

Comment (wdoug) [reply client/src/components/Register.js]: Anyway, we could make that change as a follow up if we want to.

File: client/src/components/Register.js
Comment (wdoug) [inline client/src/components/Register.js]: The `contentLabel` is for the html `aria-label`. Let's change this to say `"Terms"` or something else that is descriptive of the modal.

**Comment by wdoug** [comment]
@fredmon3 I'm requesting two small changes

**Comment by darrenrichmond** [comment]
I have verified that this PR does pop up a modal. I'll let someone else merge, since it looks like some tech issues are going on.

**Comment by wdoug** [comment]
Let's spend a minute tomorrow and see if we can make this keyboard accessible, I thought just setting the `tabindex` to 0 would do it, but that is not the case. If it takes more than a short period of time, or if we are waiting on this for something else we can go ahead and merge it.

**Review by wdoug — APPROVED** [review]
:shipit: Hawt!

**Pull request merged**
